src/provider/handler/mappers/postMapper.js

```diff
@@ -1,36 +1,50 @@
-export function mapPost(post) {
-  let result = {
-    type: {
-      value: 'article'
-    },
-    media_group: [],
-    extensions: {},
-    content: {},
-    link: {}
-  };
-
-  result.id = post.id;
-  result.title = (post.title && post.title.rendered)?post.title.rendered:'';
-  result.publish = post.date;
+export function mapPost(mediaItems) {
+  return function(post) {
+    const result = {
+      type: {
+        value: 'article'
+      },
+      id: post.id,
+      title: post.title && post.title.rendered ? post.title.rendered : '',
+      publish: post.date,
+      media_group: [],
+      extensions: {},
+      content: {},
+      link: {}
+    };
 
-  //the post's link that will be used when a user shares the post
-  result.link = {
-    type: 'text/html',
-    rel: 'alternate',
-    href: post.link
-  };
+    //the post's link that will be used when a user shares the post
+    result.link = {
+      type: 'text/html',
+      rel: 'alternate',
+      href: post.link
+    };
 
-  //article's media id, to be fetched later
-  if (post.featured_media) {
-    result.featured_media = post.featured_media;
-  }
+    //if we can find the post's media id then let's add it's url to our media_group
+    mediaItems.forEach(mediaItem => {
```

Comment:
every time you are using the `forEach` / `array.push()` combo, you can usually either use `map` or `reduce` function

I would do something like : 
```
const mediaGroups = mediaItems
   .filter(removeItemsYouDontWant)
   .map(transformMediaItemObject)

```